lib/unique_user_events/service.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# Top-level module for Unique User Metrics system
+#
+# This module provides the main entry point for logging unique user events
+# for analytics purposes. Events are tracked only once per user and include
+# automatic DataDog metric increments for new events.
+#
+# @example Basic usage
+#   UniqueUserEvents.log_event(user_id: current_user.uuid, event_name: 'mhv_landing_page_accessed')
+#
+# @example Check if event exists
+#   UniqueUserEvents.event_logged?(user_id: current_user.uuid, event_name: 'secure_messages_accessed')
+module UniqueUserEvents
+  # Service class for MHV Unique User Metrics
+  #
+  # This service provides the core business logic for logging and checking unique user events
+  # for MHV Portal analytics. It integrates with the MHVMetricsUniqueUserEvent model and provides
+  # DataDog metric increments for new events.
+  class Service
+    # StatsD key prefix for unique user metrics
+    STATSD_KEY_PREFIX = 'uum.unique_user_metrics'
+
+    # Log a unique user event with DataDog metrics
+    #
+    # @param user_id [String] UUID of the authenticated user
+    # @param event_name [String] Name of the event being logged
+    # @return [Boolean] true if new event was logged, false if already existed
+    def self.log_event(user_id:, event_name:)
+      # Record the event to database/cache
+      new_event_created = MHVMetricsUniqueUserEvent.record_event(
```

Comment:
Note: We do "optimistic insertion" here, so we do only one operation on the database.